### PR TITLE
Use gh user/pass credentials for API requests

### PR DESCRIPTION
**Pull request by JameelB**
Use Github user/password credentials for authenticating github API requests instead of the token.

### CODE REVIEW

File: src/org/integr8ly/GitHubUtils.groovy [hunk 1]
```diff
@@ -17,22 +17,20 @@ import org.kohsuke.github.PagedIterable
...
  * @param httpMethod - HTTP method to be used for the request
  * @param requestBody - HTTP request body
- * @param ghApiTokenID - The credentials id where the github api token is stored in Jenkins
+ * @param credentialsID - The credentials id where the github user/password is stored in Jenkins
  * @param customHeaders - Additional headers to be used for the request
  * @param validResponseCodes - Valid HTTP response codes
  * @returns HTTP response
  */
-def ghApiRequest(String endpoint, String httpMethod = 'GET', String requestBody = null, String ghApiTokenID, customHeaders = [], validResponseCodes = '100:399') {
+def ghApiRequest(String endpoint, String httpMethod = 'GET', String requestBody = null, String credentialsID, validResponseCodes = '100:399') {
```
Comment (mikenairn) [inline src/org/integr8ly/GitHubUtils.groovy]: I think you could have left "customHeaders" as it was tbh. Was doing no harm and is meant to allow you to easily re-use this method in the case where you need to set custom headers.

File: src/org/integr8ly/GitHubUtils.groovy [hunk 2]
```diff
@@ -17,22 +17,20 @@ import org.kohsuke.github.PagedIterable
...
-        customHeaders.plus(["name": "Authorization", "value": "token ${env.gitToken}"])
-    }
 
     def response = httpRequest httpMode: httpMethod,
+                                authentication: credentialsID,
                                 contentType: 'APPLICATION_JSON',
                                 consoleLogResponseBody: true, 
                                 requestBody: requestBody,
-                                customHeaders: customHeaders,
+                                customHeaders: [],
```
Comment (odra) [inline src/org/integr8ly/GitHubUtils.groovy]: do we need this line at all?

Comment (mikenairn) [reply src/org/integr8ly/GitHubUtils.groovy]: Not if we're not allowing it be set as one of the parameters.

Comment (JameelB) [reply src/org/integr8ly/GitHubUtils.groovy]: I've put back the customHeaders param now, should just default to an empty array

**Review by mikenairn — APPROVED** [review]

**Pull request merged**